### PR TITLE
32 MachineServiceImpl

### LINKED ISSUE #32: MachineServiceImpl
Implements MachineService Interface. 
Uses ScheduleService and the ConnectionRepository

The component is responsible for handling the connections to machines.
It handles the control of the machine by the back end.

### PR DESCRIPTION
**Pull request by Kaszz**
Closes #32

## Description
Add MachineServiceImpl class and an AutoBrewerRunner class which is responsible for running the autobrewer.

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes, if necessary.
- [x] All tests pass (existing and potential new).
- [ ] Sonar check has passed
- [x] No codesmells or bugs

## Other Relevant Information
New pull request, feel free to give the same feedback if we didn't fix it properly. Lots of stuff was split out so that it should be more manageable now.

### CODE REVIEW

**Review by Wylardt — APPROVED** [review]

File: BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Service("machine")
```
Comment (Kaszz) [inline BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java:27]: skal vi fjerne navnet?

Comment (Lawst96) [reply BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java:27]: Might as well. Der findes kun den her implementering af interfacet.

File: BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+public class MachineServiceImpl implements IMachineService {
+    static Logger logger = Logger.getLogger(MachineServiceImpl.class.getName());
```
Comment (Lawst96) [inline BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java:29]:
```suggestion
    private static Logger logger = Logger.getLogger(MachineServiceImpl.class.getName());
```
This is still default.

File: BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        }
+    }
+
+    /**
+     * Takes a command and sends it to the desired machine
+     *
+     * @param command      The {@code command} that should be sent to the machine
+     * @param connectionID The ID of the machine that should receive the {@code command}
+     * @return Returns {@code true} if the control {@code command} was successfully sent, returns {@code false} if the control {@code command} failed
+     */
```
Comment (Lawst96) [inline BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java]:
```suggestion
    /**
     * {@inheritdoc}
     */
```
Either inherit the javadoc from the interface or just remove it entirely, no reason to copy it.

**Review by Lawst96 — APPROVED** [review]

**Review by Kechr — APPROVED** [review]

File: BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/AutobrewRunner.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package com.brewmes.machine;
+
+import com.brewmes.common.entities.ScheduledBatch;
+import com.brewmes.common.services.IMachineService;
+import com.brewmes.common.services.IScheduleService;
+import com.brewmes.common.services.ISubscribeService;
+import com.brewmes.common.util.Command;
+import com.brewmes.common.util.MachineState;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (Kaszz) [inline BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/AutobrewRunner.java:9]:
```suggestion
```

**Review by dottore1 — APPROVED** [review]

**Pull request merged**